### PR TITLE
Di 1289 restore fuctionality to pass dir path as an input

### PR DESCRIPTION
**Pull request by Yu-jinKim**
This change is [Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_conductor/155)

## Summary by CodeRabbit

- **New Features**
	- Updated application version to 2.1.0.
	- Introduced a new function to retrieve job output directories.
	- Added new test cases for validating output directory handling and error conditions.

- **Bug Fixes**
	- Enhanced error handling for job outputs and fastq file validation.

- **Documentation**
	- Improved logging for output retrieval processes.

- **Refactor**
	- Streamlined test structure for better maintainability.
	- Updated input handling logic to integrate job outputs effectively.

### CODE REVIEW

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 1]
```diff
@@ -5,6 +5,7 @@
 
 import pytest
 
+import utils
```
[the remark this answers is not in the document]
Comment (jethror1) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:8]: agreed

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 2]
```diff
@@ -598,88 +599,94 @@
...
-        "sample_name_prefix": "INPUT-SAMPLE-PREFIX",
-        "sample_name": "INPUT-SAMPLE-NAME",
-        "output_path": "INPUT-analysis_1-out_dir",
-        "run_sample_sheet": {"$dnanexus_link": "INPUT-SAMPLESHEET"},
-    }
+    @pytest.fixture
+    def other_inputs(self, monkeypatch):
+        monkeypatch.setattr(
+            "utils.manage_dict.get_job_out_folder",
+            lambda x: "/out_dir/dir1",
```
[the remark this answers is not in the document]
Comment (jethror1) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:606]: lambda functions don't require an arg, if it's not needed can just be dropped:
```
>>> a = lambda: 2 * 3
>>> a()
6
```

File: resources/home/dnanexus/run_workflows/utils/manage_dict.py [hunk 3]
```diff
@@ -364,6 +366,37 @@ def add_other_inputs(
...
+        )
+
+        if not analysis_job_id:
+            raise KeyError(
+                (
+                    "Error trying to parse output directory to input dict."
+                    f"\nNo output directory found for given input: {out_dir}\n"
+                    "Please check config to ensure job input is in the "
+                    "format: INPUT-analysis_[0-9]-out_dir"
+                )
```
[the remark this answers is not in the document]
Comment (jethror1) [reply resources/home/dnanexus/run_workflows/utils/manage_dict.py:387]: are you using Black here? has it added the extra parentheses?

**Review by jethror1 — COMMENTED** [review]
> Reviewed 3 of 5 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_conductor/155)* status: 3 of 5 files reviewed, 13 unresolved discussions (waiting on @coderabbitai[bot] and @Yu-jinKim)

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py]: Done.

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 4]
```diff
@@ -598,88 +599,94 @@
...
-        "eggd_multiqc_config_file": {
-            "$dnanexus_link": "file-G0K191j433Gv6JG63b43z8Gy"
-        },
-        "sample_name_prefix": "INPUT-SAMPLE-PREFIX",
-        "sample_name": "INPUT-SAMPLE-NAME",
-        "output_path": "INPUT-analysis_1-out_dir",
-        "run_sample_sheet": {"$dnanexus_link": "INPUT-SAMPLESHEET"},
-    }
+    @pytest.fixture
+    def other_inputs(self, monkeypatch):
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py]: Done.

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py]: Nah the function being mocked does take an argument so if no argument is passed to the lambda function i get this:
```
TypeError: <lambda>() takes 0 positional arguments but 1 was given
```

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 5]
```diff
@@ -598,88 +599,94 @@
...
+            parent_out_dir=parent_out_dir,
+            project_id=dx_project_id,
+            project_name=dx_project_name,
+            sample="my_sample_with_a_long_name",
+            sample_prefix="my_sample",
+            job_outputs_dict=job_outputs,
+        )
 
-    def test_adding_sample_name(self):
+    def test_adding_sample_name(self, other_inputs):
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:645]: Will not do

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 6]
```diff
@@ -598,88 +599,94 @@
...
         """
         Test for finding INPUT-SAMPLE-NAME and replacing with sample name
         """
         assert (
-            self.output["sample_name"] == "my_sample_with_a_long_name"
+            other_inputs["sample_name"] == "my_sample_with_a_long_name"
         ), "INPUT-SAMPLE-NAME not correctly replaced"
 
-    def test_adding_sample_prefix(self):
+    def test_adding_sample_prefix(self, other_inputs):
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:653]: Will not do

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 7]
```diff
@@ -598,88 +599,94 @@
...
         """
         Test for finding INPUT-SAMPLE-PREFIX and replacing with sample prefix
         """
         assert (
-            self.output["sample_name_prefix"] == "my_sample"
+            other_inputs["sample_name_prefix"] == "my_sample"
         ), "INPUT-SAMPLE-PREFIX not correctly replaced"
 
-    def test_adding_project_id(self):
+    def test_adding_project_id(self, other_inputs):
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:661]: Will not do

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 8]
```diff
@@ -598,88 +599,94 @@
...
         Test for finding INPUT-dx_project_id and replacing with project_id
         from args Namespace object
         """
         assert (
-            self.output["my_project_id"] == "project-12345"
+            other_inputs["my_project_id"] == "project-12345"
         ), "INPUT-dx_project_id not correctly replaced"
 
-    def test_adding_project_name(self):
+    def test_adding_project_name(self, other_inputs):
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:670]: Will not do

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 9]
```diff
@@ -598,88 +599,94 @@
...
         Test for finding INPUT-dx_project_name and replacing with project_name
         from args Namespace object
         """
         assert (
-            self.output["my_project_name"] == "some_analysis_project"
+            other_inputs["my_project_name"] == "some_analysis_project"
         ), "INPUT-dx_project_name not correctly replaced"
 
-    def test_adding_parent_out_dir(self):
+    def test_adding_parent_out_dir(self, other_inputs):
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:679]: Will not do

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 10]
```diff
@@ -598,88 +599,94 @@
...
         Test for finding INPUT-parent_out_dir and replacing with parent output
         directory from args Namespace object
         """
         assert (
-            self.output["all_analysis_output"] == "some_assay-220930-1200"
+            other_inputs["all_analysis_output"] == "some_assay-220930-1200"
         ), "INPUT-parent_out_dir not correctly replaced"
 
-    def test_adding_samplesheet(self):
+    def test_adding_samplesheet(self, other_inputs):
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:688]: Will not do

File: resources/home/dnanexus/run_workflows/tests/test_manage_dict.py [hunk 11]
```diff
@@ -689,9 +696,19 @@
         }
 
         assert (
-            self.output["run_sample_sheet"] == correct_samplesheet
+            other_inputs["run_sample_sheet"] == correct_samplesheet
         ), "Samplesheet not correctly parsed to input dict"
 
+    def test_adding_analysis_1_out_dir(self, other_inputs):
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/tests/test_manage_dict.py:701]: Will not do

File: resources/home/dnanexus/run_workflows/utils/manage_dict.py [hunk 12]
```diff
@@ -279,6 +280,7 @@
     project_name,
     sample=None,
     sample_prefix=None,
+    job_outputs_dict=None,
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/utils/manage_dict.py]: Will not do

File: resources/home/dnanexus/run_workflows/utils/manage_dict.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/utils/manage_dict.py]: i was, i removed them and Black didn't add them back :man_shrugging:

File: resources/home/dnanexus/run_workflows/utils/manage_dict.py [hunk 13]
```diff
@@ -364,6 +366,37 @@
...
+        )
+
+        if not analysis_job_id:
+            raise KeyError(
+                (
+                    "Error trying to parse output directory to input dict."
+                    f"\nNo output directory found for given input: {out_dir}\n"
+                    "Please check config to ensure job input is in the "
+                    "format: INPUT-analysis_[0-9]-out_dir"
+                )
```
[the remark this answers is not in the document]
Comment (Yu-jinKim) [reply resources/home/dnanexus/run_workflows/utils/manage_dict.py]: Will not do

**Review by Yu-jinKim — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_conductor/155)* status: 3 of 5 files reviewed, 13 unresolved discussions (waiting on @coderabbitai[bot] and @jethror1)

**Review by jethror1 — APPROVED** [review]
> Reviewed 1 of 5 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_conductor/155)* status: all files reviewed, 17 unresolved discussions (waiting on @Yu-jinKim)

**Review by jethror1 — APPROVED** [review]
> *[Reviewable](https://reviewable.io/reviews/eastgenomics/eggd_conductor/155)* status: all files reviewed, 17 unresolved discussions (waiting on @Yu-jinKim)

**Pull request merged**